dataprofiler/tests/profilers/test_categorical_column_profile.py

```diff
@@ -225,11 +251,80 @@ def test_categorical_merge(self):
         self.assertEqual(profile3.is_match, False)
         self.assertEqual(profile3.unique_ratio, 16 / 33)
 
+        report = profile3.profile
+        report.pop("times")
+        expected_profile = dict(
+            categorical=False,
+            statistics=dict([
+                ('unique_count', 16),
+                ('unique_ratio', 16 / 33),
+            ]),
+        )
+        self.assertEqual(report, expected_profile)
+
         # Check is_match and unique_ratio if the sample size was large
         profile3.sample_size = 1000
         self.assertEqual(profile3.is_match, True)
         self.assertEqual(profile3.unique_ratio, 16 / 1000)
 
+        report = profile3.profile
+        report.pop("times")
+        report_categories = report['statistics'].pop('categories')
+        report_count = report['statistics'].pop('categorical_count')
+        expected_profile = dict(
+            categorical=True,
+            statistics=dict([
+                ('unique_count', 16),
+                ('unique_ratio', 16 / 1000)
+            ]),
+        )
+        self.assertEqual(report, expected_profile)
+        self.assertCountEqual(report_categories, ['abcd', 'aa', '2', np.nan,
+                                                  '4', 'b', '3', 'dfd', 'ee',
+                                                  'ff', 'nan', 'None', '1',
+                                                  'gg', 'null', 'NaN'])
+        expected_dict = dict({'aa': 5, '2': 4, 'abcd': 4, 'b': 3, np.nan: 2})
+        self.assertEqual(len(report_count), len(expected_dict))
+        self.assertEqual(report_count['aa'], expected_dict['aa'])
+        self.assertEqual(report_count['2'], expected_dict['2'])
+        self.assertEqual(report_count['abcd'], expected_dict['abcd'])
+        self.assertEqual(report_count['b'], expected_dict['b'])
+
+    def test_categorical_column_is_counted(self):
+       df_categorical = pd.Series([
+           "a", "a", "a", "b", "b", "b", "b", "c", "c", "c", "c", "c",
+       ])
+       profile = CategoricalColumn(df_categorical.name)
+
+       # Check that _categories counts in profile is empty after init.
+       report = profile.profile
+       counts = report["statistics"]["categorical_count"]
+       self.assertEqual(0, len(counts))
+       profile.update(df_categorical)
+
+       report = profile.profile
+       counts = report["statistics"]["categorical_count"]
+
+       # Check that _categories counts is correct after update function
+       self.assertEqual(counts['a'], 3)
+       self.assertEqual(counts['b'], 4)
+       self.assertEqual(counts['c'], 5)
+
+       df_categorical2 = pd.Series([
+           "a", "a", "d", "d"
+       ])
+       profile2 = CategoricalColumn(df_categorical2.name)
+       profile2.update(df_categorical2)
```

Comment:
can we add test for categorical count for profile2?

Reply:
test being removed.